tests/openai/test_openai_protocol.py

```diff
@@ -0,0 +1,189 @@
+    )
+    assert completion.model == openai_testing_model
+    assert completion.model
+    # assert completion.choices[0].text == "Hello world"
+
+
+def chat(openai_testing_model):  # noqa: F811
+    # create a chat completion
+    chat_completion = client.chat.completions.create(
+        model=openai_testing_model,
+        messages=[{"role": "user", "content": "Hello world"}],
+        top_p=1,
+    )
+    assert chat_completion
+    assert chat_completion.usage
+    assert chat_completion.id
+    assert isinstance(chat_completion.choices, list)
+    assert chat_completion.choices[0].message.content
+
+
+def completions_bad_request(openai_testing_model):  # noqa: F811
+    with pytest.raises(openai.BadRequestError) as exc_info:
+        client.completions.create(
+            model=openai_testing_model,
+            prompt="Hello world",
+            temperature=-0.1,
+        )
+    assert "temperature" in str(exc_info.value)
+
+
+def chat_bad_request(openai_testing_model):  # noqa: F811
+    with pytest.raises(openai.BadRequestError) as exc_info:
+        client.chat.completions.create(
+            model=openai_testing_model,
+            messages=[{"role": "user", "content": "Hello world"}],
+            temperature=-0.1,
+        )
+    assert "temperature" in str(exc_info.value)
+
+
+def completions_stream(openai_testing_model):  # noqa: F811
+    i = 0
+    for completion in client.completions.create(
+        model=openai_testing_model, prompt="Hello world", stream=True, top_p=1
+    ):
+        i += 1
+        assert completion
+        assert completion.id
+        assert isinstance(completion.choices, list)
+        assert isinstance(completion.choices[0].text, str)
+    assert i > 4
+
+
+def chat_stream(openai_testing_model):  # noqa: F811
+    i = 0
+    for chat_completion in client.chat.completions.create(
+        model=openai_testing_model,
+        messages=[{"role": "user", "content": "Hello world"}],
+        stream=True,
+        temperature=0.4,
+        frequency_penalty=0.02,
+        top_p=1,
+    ):
+        if i == 0:
+            assert chat_completion
+            assert chat_completion.id
+            assert isinstance(chat_completion.choices, list)
+            assert chat_completion.choices[0].delta.role
+        else:
+            assert chat_completion
+            assert chat_completion.id
+            assert isinstance(chat_completion.choices, list)
+            assert chat_completion.choices[0].delta == {} or hasattr(
+                chat_completion.choices[0].delta, "content"
+            )
+        i += 1
+    assert chat_completion
+    assert chat_completion.id
+    assert isinstance(chat_completion.choices, list)
+    assert chat_completion.choices[0].delta == {} or hasattr(
+        chat_completion.choices[0].delta, "content"
+    )
+    assert chat_completion.choices[0].model_fields["finish_reason"]
+    assert i > 4
+
+
+def completions_stream_bad_request(openai_testing_model):  # noqa: F811
```

Comment:
We can remove 4 methods here starting with `completions_` as they are not used.

Reply:
ok